Add markAsCash button with wired up action for dismissing the rter violation

BrtqKr opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
$ https://github.com/Expensify/App/issues/39541
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->

PROPOSAL:


### Tests

1. Go to the open report  - a workspace for example
2. Submit an expense and retrieve its `transactionId`
3. Go to the report screen and in the console, update the onyx entries containing the mocked card that will be used for the transaction, the given transaction and the corresponding violation. 

```
Onyx.merge('cardList', {
        1: {
            bank: 'Expensify Card',
            lastFourPAN: 3458,
        },
    });

Onyx.merge(`transactions_<transaction_id>`, {cardID: 1, merchant: 'violation test'});

Onyx.merge(`transactionViolations_<transaction_id>`, [{type: 'test', name: CONST.VIOLATIONS.RTER, data: {pendingPattern: true}}]);
```
4. The mark as cash button should appear
5. After pressing it the button and the violation message should disappear and there should be a response in the thread confirming that there was a conversion to cash 

- [x] Verify that no errors appear in the JS console

### Offline tests
As the standard ones.

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<img src="https://github.com/Expensify/App/assets/45288762/7ff35984-cf50-482e-83d0-5619fe83da91" width="200" />
<img src="https://github.com/Expensify/App/assets/45288762/94cc45d2-2e89-4c61-99d0-5258d9b356a0" width="200" />
<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<img src="https://github.com/Expensify/App/assets/45288762/7d23e7bd-d3a1-479d-96e4-5efa94fc819c" width="200" />

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<img src="https://github.com/Expensify/App/assets/45288762/628c309d-c037-4239-96a3-8ff137bf6ea8" width="200" />

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<img src="https://github.com/Expensify/App/assets/45288762/1df4c432-51bd-4e2f-9826-f738283ded13" width="200" />

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/45288762/d9c98017-860e-4574-903f-da6ccd5fefa3

https://github.com/Expensify/App/assets/45288762/bf66977f-62e9-4ecf-bd90-153b24b293f7


<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/45288762/76f21bbe-9ed9-45ca-9ce2-d70f96a8bfbe

<!-- add screenshots or videos here -->

</details>

BrtqKr committed:
add mark as cash button

BrtqKr committed:
fix button border

BrtqKr committed:
Merge remote-tracking branch 'upstream/yuwen-dismissAction' into brtqkr/mark-as-cash

BrtqKr committed:
wire up action

BrtqKr committed:
fix report id

kosmydel commented on src/pages/home/report/ReportActionsList.tsx:
```diff
@@ -188,6 +188,7 @@ function ReportActionsList({
     const hasFooterRendered = useRef(false);
     const lastVisibleActionCreatedRef = useRef(report.lastVisibleActionCreated);
     const lastReadTimeRef = useRef(report.lastReadTime);
+    // Single MoneyRequest
```
Do we need this comment?

BrtqKr replied:
It is showing as a change, because ola removed it in her PR and this branch was based on it

kosmydel commented on src/components/MoneyRequestHeaderStatusBar.tsx:
```diff
@@ -35,12 +40,16 @@ function MoneyRequestHeaderStatusBar({title, description, shouldShowBorderBottom
-                    badgeStyles={styles.ml0}
-                    error={danger}
-                />
+            <View style={additionalViewStyle ?? [styles.mr3]}>
```
NAB
I think that `additional` in the variable name might be misleading. We pass an additional style, which replaces the default one. Instead, we might consider overriding it like this:
```suggestion
            <View style={[styles.mr3, additionalViewStyle]}>
```

kosmydel commented on src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx:
```diff
@@ -319,6 +322,30 @@ function MoneyRequestPreviewContent({
+                                                    width={variables.iconSizeExtraSmall}
+                                                    fill={theme.textSupporting}
+                                                />
+                                                <Text style={[styles.textLabel, styles.colorMuted, styles.ml1, styles.amountSplitPadding]}>{translate('iou.receiptScanInProgress')}</Text>
```
NAB: We can consider creating a new style for it.

BrtqKr replied:
I would check if any other use cases appear in the first place and apply it separately, but yeah that's probably a good idea

kosmydel commented on src/components/MoneyRequestHeader.tsx:
```diff
@@ -116,6 +126,21 @@ function MoneyRequestHeader({
         }
     };
 
+    const getPendingType = () => {
+        if (TransactionUtils.isExpensifyCardTransaction(transaction) && TransactionUtils.isPending(transaction)) {
+            return {pendingType: 'PENDING', pendingTitle: translate('iou.pending'), pendingDescription: translate('iou.transactionPendingText')};
+        }
+        if (isScanning) {
+            return {pendingType: 'SCANNING', pendingTitle: ReceiptScan, pendingDescription: translate('iou.receiptScanInProgressDescription')};
+        }
+        if (TransactionUtils.hasPendingRTERViolation(TransactionUtils.getTransactionViolations(transaction?.transactionID ?? '', transactionViolations))) {
+            return {pendingType: 'RTER', pendingTitle: Expensicons.Hourglass, pendingDescription: translate('iou.pendingMatchWithCreditCardDescription')};
+        }
+        return {};
+    };
```
To consider: adding `useCallback`.

BrtqKr replied:
- it's based on the transaction, which is one of the key elements that change in this components props, so probably there wouldn't be any major benefits coming from this change
- probably useMemo would be more suitable, but as I stated, there aren't any major benefits and since there aren't any issues either we should avoid it [as stated in the docs](https://react.dev/reference/react/useMemo#should-you-add-usememo-everywhere)

kosmydel commented on src/libs/actions/Transaction.ts:
```diff
@@ -264,4 +271,52 @@ function clearError(transactionID: string) {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${transactionThreadReportID}`,
+                value: {
+                    [optimisticReportAction.reportActionID]: null,
```
Just want to make sure: which offline pattern do we follow here? Is it B (Optimistic WITH Feedback Pattern)?

BrtqKr replied:
it's from the other branch that has already been merged, but @yuwenmemon probably has more context regarding this action

kosmydel reviewed (commented):
A few minor questions

yuwenmemon commented:
@BrtqKr Merge main to simplify the diff a bit!

yuwenmemon commented:
@BrtqKr Quick bump on this! I believe all pre-requisite PRs have been merged

BrtqKr committed:
Merge remote-tracking branch 'origin/main' into brtqkr/mark-as-cash

dubielzyk-expensify reviewed (dismissed):
Looks good to me from a visual perspective. cc @Expensify/design for visibility

dubielzyk-expensify commented:
Oh wow. What a miss on my end. Yep most definitely. Should look like this:

![image](https://github.com/Expensify/App/assets/145416827/ea2a2f43-f9b1-4411-8c98-f3b6c937b8d0)

shawnborton commented:
Yup, exactly!

BrtqKr committed:
handle button ui for the web breakpoints

BrtqKr committed:
Merge remote-tracking branch 'origin/main' into brtqkr/mark-as-cash

BrtqKr commented:
ok, it should be fine now
<img src="https://github.com/Expensify/App/assets/45288762/66f072ea-fcf5-453b-90fb-d820ecf0b420" width="200" />

<img width="200" alt="Screenshot 2024-05-17 at 10 12 45" src="https://github.com/Expensify/App/assets/45288762/a448e2f9-1749-4264-a328-12eac8f173fd">

<img src="https://github.com/Expensify/App/assets/45288762/bf457d21-05c0-4952-941f-a29abb6e9e81" width="800" />

shawnborton commented:
I could have sworn on mobile, we show the green button above the status - does that sound correct?

c3024 commented:
For oneTransactionReports the header is `MoneyReportHeader` and there is no code in this header for `Mark as Cash` button. Is this as per design?


<img width="1437" alt="Screenshot 2024-05-17 at 5 51 05 PM" src="https://github.com/Expensify/App/assets/102477862/0441ccf0-eebd-45b1-a605-0b8c04437099">

shawnborton commented:
@dannymcclain oh boy... @puneetlath do you have a screenshot handy of what it's like to approve a report on NewDot in mobile, where you have a green button and next steps? I could have sworn the green button shows above the next steps... and if so, we should stay consistent here too.

shawnborton commented:
> For oneTransactionReports the header is MoneyReportHeader and there is no code in this header for Mark as Cash button. Is this as per design?

Good catch, we should totally add it here too I would think. @trjExpensify do you agree?

puneetlath commented:
>@puneetlath do you have a screenshot handy of what it's like to approve a report on NewDot in mobile, where you have a green button and next steps? I could have sworn the green button shows above the next steps... and if so, we should stay consistent here too.

Here ya go

![IMG_D8B6A16491A0-1](https://github.com/Expensify/App/assets/1371996/b635edb9-e52b-456b-b39b-cd9e6af27a69)

dannymcclain commented:
@puneetlath thanks!

@shawnborton I agree we should keep the placement consistent.

yuwenmemon reviewed (changes requested):
Code LGTM but we're missing this button on the "Single Expense" Report View:

https://github.com/Expensify/App/assets/4741899/c6757a2f-d8ea-4067-b683-8640fec6ab22

BrtqKr committed:
change conditon for layout, add mark as cash to moneyReportHeader, make button appear only for open reports

BrtqKr commented:
@yuwenmemon  I've moved the button to the top, added it to the report header, and made it for draft only, since according to what we've discussed [in here](https://swmansion.slack.com/archives/C06ML6X0W9L/p1715282600935749?thread_ts=1715094520.569059&cid=C06ML6X0W9L) the action wouldn't go through without the isDraft flag. I believe that should be everything

dannymcclain commented:
>Should this report action be here in the report with all transactions?

Hmm that seems weird to me. I would expect it to be in the transaction thread, not the report thread. (Like if you change the category of an expense, that system message shows up in the expense thread.) I would also expect it to have the avatar/name of the person who marked it as cash. @JmillsExpensify @shawnborton does that sound right to you?

>Another question unrelated to the above, should an RBR be shown for this violation on report preview, LHN and in the transaction report (not sure which field that would be.)?

I think @JmillsExpensify would know better, but I feel like the preview _should_ be showing an RBR/error message instead of "Unknown Merchant"... Unless of course someone set the merchant to `Unknown Merchant` deliberately. 😅

dannymcclain commented:
>I agree it should be in the transaction thread. As for the avatar, I think it follows the same logic as when the same user sends multiple messages consecutively where we skip the avatar if you send a bunch of messages in a row.

Ah that makes sense. So since this is (wrongly) showing up in the report thread, it's piggy-backing off of the request message. Once we get it in the expense thread like it's supposed to be, that issue should go away. 🤞

yuwenmemon commented:
Hmmm yeah @BrtqKr That seems to be stemming from some sort of optimistic action we're adding to the parent report because once you sign in and out, those messages disappear:

https://github.com/Expensify/App/assets/4741899/fa49db82-f579-4129-b030-404991ca6a3a

![Screenshot 2024-05-20 at 11 56 56 AM](https://github.com/Expensify/App/assets/4741899/d663bf6f-140c-4100-939a-f4b7804e7719)

BrtqKr commented:
@yuwenmemon isn't it, because we're using a random id for the messaging optimistic update? Maybe Onyx isn't able to relate the success id and the random one generated in the action and we should be removing it in the `successData` 

```
            // Dismiss action with a temporary id
            {
                onyxMethod: Onyx.METHOD.MERGE,
                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${transactionThreadReportID}`,
                value: {
                    [optimisticReportAction.reportActionID]: null,
                },
            },
```

https://github.com/Expensify/App/assets/45288762/157ed4ae-68dd-45b7-8a2c-da1a801fcddb

yuwenmemon commented on src/components/MoneyRequestHeader.tsx:
```diff
@@ -231,7 +240,32 @@ function MoneyRequestHeader({
+                            text={translate('iou.markAsCash')}
+                            style={[styles.p0]}
+                            onPress={() => {
+                                TransactionActions.markAsCash(transaction?.transactionID ?? '', transaction?.reportID ?? '');
```
@BrtqKr I think this is the source of your issue. We should be using the transactionThreadID here, not the report ID of the transaction.

One can see this here:
<img width="1650" alt="Screenshot 2024-05-21 at 11 48 06 AM" src="https://github.com/Expensify/App/assets/4741899/39e6f2f6-3b0f-46cc-ba0c-3f8ae3574bc4">
<img width="659" alt="Screenshot 2024-05-21 at 11 48 13 AM" src="https://github.com/Expensify/App/assets/4741899/6eb89c06-6515-4221-94c8-0c8ca6543741">

yuwenmemon commented:
@BrtqKr I don't think it's that, no. I commented on the PR on the line that I believe to be the culprit. Essentially we're passing the transaction's reportID to `TransactionActions.markAsCash` when we should be passing the Transaction Thread ReportID instead.

BrtqKr committed:
add threadId to the action

JmillsExpensify commented:
Quickly confirming that yes, this should appear in the transaction thread.

yuwenmemon reviewed (dismissed):
LGTM!

yuwenmemon commented:
@c3024 Can you prioritize your review of this PR ASAP?

c3024 commented on src/components/MoneyRequestHeader.tsx:
```diff
@@ -231,7 +244,32 @@ function MoneyRequestHeader({
+                            text={translate('iou.markAsCash')}
+                            style={[styles.p0]}
+                            onPress={() => {
+                                TransactionActions.markAsCash(transaction?.transactionID ?? '', transactionThreadReportID ?? '');
```
```suggestion
                                TransactionActions.markAsCash(transaction?.transactionID ?? '', report?.reportID ?? '');
```

c3024 commented on src/components/MoneyRequestHeader.tsx:
```diff
@@ -231,7 +244,32 @@ function MoneyRequestHeader({
+                            text={translate('iou.markAsCash')}
+                            style={[styles.w100, styles.pr0]}
+                            onPress={() => {
+                                TransactionActions.markAsCash(transaction?.transactionID ?? '', transaction?.reportID ?? '');
```
```suggestion
                                TransactionActions.markAsCash(transaction?.transactionID ?? '', report?.reportID ?? '');
```

I think it is better to extract this to a local function because it is used twice and in the last commit only one instance was updated.

c3024 commented on src/components/MoneyRequestHeader.tsx:
```diff
@@ -147,7 +156,11 @@ function MoneyRequestHeader({
         }
         if (TransactionUtils.hasPendingRTERViolation(TransactionUtils.getTransactionViolations(transaction?.transactionID ?? '', transactionViolations))) {
-            return {title: getStatusIcon(Expensicons.Hourglass), description: translate('iou.pendingMatchWithCreditCardDescription'), shouldShowBorderBottom: true};
+            return {
```
This should be moved above the `isScanning` message. That is how it is in oneTransactionReports. When there is an RTER violation that message is shown instead of the 'being scanned' message.

BrtqKr committed:
extract mark as cash

BrtqKr committed:
cleanup

c3024 commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

c3024 commented:
NAB: `MoneyReportHeader` props do not include `transaction` or `transactionViolations` so it does not appear to rerender immediately on `Onyx.merge` `transaction` and `transactionViolations`. But in actual use cases, I think backend sends more info than just `transaction` and `transactionViolations` so it should rerender immediately.  

Also, `MoneyReportHeader` here is used only for oneTransactionReports, I think it is fine to not subscribe to these changes and avoid rerenders.

c3024 reviewed (approved):
LGTM!

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9256014117) to staging by https://github.com/yuwenmemon in version: 1.4.76-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

kavimuru commented:
@BrtqKr Should the expense be a credit card expense? Should this be created in OldDot?
